extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/configuration_validation/ClassWithConfigProperties.java

```diff
@@ -0,0 +1,69 @@
+package io.quarkus.hibernate.validator.runtime.configuration_validation;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.enterprise.inject.spi.CDI;
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import javax.validation.Validator;
+
+/**
+ * Holds the class and the configuration properties i.e class fields annotated with
+ * {@link org.eclipse.microprofile.config.inject.ConfigProperty}
+ */
+public class ClassWithConfigProperties {
+    private final Class<?> aClass;
+    private final List<String> configurationProperties;
+    static List<ClassWithConfigProperties> classWithConfigurationProperties;
+
+    ClassWithConfigProperties(Class<?> aClass, List<String> configurationProperties) {
+        this.aClass = aClass;
+        this.configurationProperties = configurationProperties;
+    }
+
+    /**
+     * Throw an {@link ConstraintViolationException} when a configuration property is not valid
+     */
+    void validate(Validator validator) {
+        for (Object obj : CDI.current().select(aClass)) {
+            for (String property : configurationProperties) {
+                final Set<ConstraintViolation<Object>> constraintViolations = validator.validateProperty(obj, property);
+                if (!constraintViolations.isEmpty()) {
+                    throw new ConstraintViolationException(
+                            constraintViolationMessageForProperty(property, constraintViolations), constraintViolations);
+                }
+            }
+        }
+    }
+
+    private String constraintViolationMessageForProperty(String property,
+            Set<ConstraintViolation<Object>> constraintViolations) {
+        final StringBuilder message = new StringBuilder(String.format(
+                "The configuration property \"%s\" in class \"%s\" is invalid. The following constraints violations found:",
+                property, aClass.getName()));
+
+        for (ConstraintViolation<Object> constraintViolation : constraintViolations) {
+            message.append(String.format("\n %s %s", constraintViolation.getInvalidValue(), constraintViolation.getMessage()));
+        }
+
+        return message.toString();
+    }
+
+    @Override
+    public String toString() {
+        return "ClassWithConfigProperties{" +
+                "aClass=" + aClass +
+                ", configurationProperties=" + configurationProperties +
+                '}';
+    }
+
+    public static void initialize(Map<Class<?>, List<String>> configClassWithProperties) {
```

Comment:
I personally don't like this idiom of using static methods to initialize things, but 1) I know I am a minority since the alternative of generating bytecode is harder, 2) it's already being done in various parts of the code.

Reply:
To be honest, I also wanted to try the class generation kung-fu, only done a couple of time so I dod not have the "black" belt. I leaned toward this option as it maintainable but If we both think that  generating bytecode is the way to go, I'll be happy to  give it a try.  

A few options I was planning to explore:
-  For each `ClassWithConfigProperties` construct a "bridge" class with a `@ConfigProperties` and add all proper annotations for each field so that they will be validated automatically since we have bean validation already in place for `ConfigProperties` class. 
- Find a way to hook in `ClassConfigPropertiesUtil` and add validations specific code for found classes (ignore fields that are not config properties).

Reply:
I wouldn't leverage `@ConfigProperties`  for this since it doesn't seem natural to generate such an ex class just to leverage validation.
If it were me implementing this, I would just generate some classes that integrate with validation in a similar way that `@ConfigProperties` does.
But I would say don't do it yet. Let's see what @gsmet thinks.